app/models/concerns/account_counters.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+module AccountCounters
+  extend ActiveSupport::Concern
+
+  included do
+    has_one :account_stat, inverse_of: :account
+    after_initialize :set_account_stat
+    after_save :save_account_stat
+  end
+
+  def statuses_count
+    account_stat&.statuses_count || 0
+  end
+
+  def statuses_count=(val)
+    (account_stat || build_account_stat).statuses_count = val
+  end
+
+  def following_count
+    account_stat&.following_count || 0
+  end
+
+  def following_count=(val)
+    (account_stat || build_account_stat).following_count = val
+  end
+
+  def followers_count
+    account_stat&.followers_count || 0
+  end
+
+  def followers_count=(val)
+    (account_stat || build_account_stat).followers_count = val
+  end
+
+  def increment_count!(key)
+    update_account_stat!(key => public_send(key) + 1)
+  end
+
+  def decrement_count!(key)
+    update_account_stat!(key => [public_send(key) - 1, 0].max)
+  end
+
+  private
+
+  def set_account_stat
+    self.account_stat = build_account_stat if account_stat.nil?
```

Comment:
per our conversation, this should handle migrating the counts from the existing record if there's no account_stat record.